stratum/hal/lib/common/yang_parse_tree_paths.cc

```diff
@@ -1047,7 +1047,7 @@ void SetUpInterfacesInterfaceEthernetConfigMacAddress(uint64 node_id,
     std::string mac_address_string = typed_val->string_val();
     ::google::protobuf::uint64 mac_address =
         YangStringToMacAddress(mac_address_string);
-    if (mac_address == 0) {
+    if (mac_address_string.compare("00:00:00:00:00:00") !=0 && mac_address == 0) {
```

Comment:
Use `==` for string comparison where the ordering does not matter.

Reply:
Cool...Good point.
I've added the test for InterfacesInterfaceEthernetConfigMacAddressOnUpdateFailure as your suggestion, thanks.